cmd/mobileproxy/mobileproxy.go

```diff
@@ -0,0 +1,151 @@
+// Package mobileproxy instantiates a Martian Proxy (go/martianproxy).
+// This package is a reference implmentation of Martian Proxy intended to
+// be cross compiled with google3/go/tools/gobind for use on Android and iOS.
+package mobileproxy
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"github.com/google/martian"
+	"github.com/google/martian/har"
+	"github.com/google/martian/httpspec"
+	mlog "github.com/google/martian/log"
+	"github.com/google/martian/martianhttp"
+	"github.com/google/martian/mitm"
+	"github.com/google/martian/verify"
+	"log"
+	"net"
+	"net/http"
+	"time"
+
+	// side-effect importing modifiers to register them with the proxy
+	_ "github.com/google/martian/body"
+	_ "github.com/google/martian/cookie"
+	_ "github.com/google/martian/fifo"
+	_ "github.com/google/martian/header"
+	_ "github.com/google/martian/martianurl"
+	_ "github.com/google/martian/method"
+	_ "github.com/google/martian/pingback"
+	_ "github.com/google/martian/priority"
+	_ "github.com/google/martian/querystring"
+	_ "github.com/google/martian/skip"
+	_ "github.com/google/martian/status"
+)
+
+// Martian is a wrapper for the initialized Martian proxy.
+type Martian struct {
+	proxy    *martian.Proxy
+	listener net.Listener
+	mux      *http.ServeMux
+}
+
+// Start runs a martian.Proxy on addr.
+func Start(proxyAddr string) (*Martian, error) {
+	return StartWithCertificate(proxyAddr, "", "")
+}
+
+// StartWithCertificate runs a proxy on addr and configures a cert for MITM
+func StartWithCertificate(proxyAddr string, cert string, key string) (*Martian, error) {
```

Comment:
As a second comment on this signature, In my experience passing in the listener you want to use as opposed to telling it what you want that listener to look like tends to work better for testing. Not sure if that would help you or not but keep it in mind as you progress further.